File: core-geo/src/main/java/com/example/core_geo/Coordinates.java
package com.example.core_geo;

import android.os.Parcelable;

import com.google.auto.value.AutoValue;

/**
 * Created by lev.novikov
 * Date: 20/12/17.
 */

@AutoValue
public abstract class Coordinates implements Parcelable {

    public abstract double x();

    public abstract double y();

    public static Coordinates create(double x, double y) {
        return new AutoValue_Coordinates(x, y);
    }
}

File: core-geo/src/main/java/com/example/core_geo/Coordinates.kt
package com.example.core_geo

import android.annotation.SuppressLint
import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

/**
 * Created by lev.novikov
 * Date: 20/12/17.
 */

@SuppressLint("ParcelCreator")
@Parcelize
data class Coordinates(val x: Double, val y: Double) : Parcelable
